src/dagmc/DagMC.cpp

```diff
@@ -0,0 +1,774 @@
+
+  EntityHandle file_set;
+  rval = MBI->create_meshset(MESHSET_SET, file_set);
+  if (MB_SUCCESS != rval)
+    return rval;
+
+  rval = MBI->load_file(cfile, &file_set, options, NULL, 0, 0);
+
+  if (MB_UNHANDLED_OPTION == rval) {
+    // Some options were unhandled; this is common for loading h5m files.
+    // Print a warning if an option was unhandled for a file that does not end in '.h5m'
+    std::string filename(cfile);
+    if (filename.length() < 4 || filename.substr(filename.length() - 4) != ".h5m") {
+      std::cerr << "DagMC warning: unhandled file loading options." << std::endl;
+    }
+  } else if (MB_SUCCESS != rval) {
+    std::cerr << "DagMC Couldn't read file " << cfile << std::endl;
+    std::string message;
+    if (MB_SUCCESS == MBI->get_last_error(message) && !message.empty())
+      std::cerr << "Error message: " << message << std::endl;
+
+    return rval;
+  }
+
+  return finish_loading();
+}
+
+// helper function to load the existing contents of a MOAB instance into DAGMC
+ErrorCode DagMC::load_existing_contents() {
+  return finish_loading();
+}
+
+// setup the implicit compliment
+ErrorCode DagMC::setup_impl_compl() {
+  // If it doesn't already exist, create implicit complement
+  // Create data structures for implicit complement
+  ErrorCode rval = GTT->setup_implicit_complement();
+  if (MB_SUCCESS != rval) {
+    std::cerr << "Failed to find or create implicit complement handle." << std::endl;
+    return rval;
+  }
+  return MB_SUCCESS;
+}
+
+// gets the entity sets tagged with geomtag 2 and 3
+// surfaces and volumes respectively
+ErrorCode DagMC::setup_geometry(Range& surfs, Range& vols) {
+  ErrorCode rval;
+
+  // get all surfaces
+  rval = GTT->get_gsets_by_dimension(2, surfs);
+  MB_CHK_SET_ERR(rval, "Could not get surfaces from GTT");
+
+  // get all volumes
+  rval = GTT->get_gsets_by_dimension(3, vols);
+  MB_CHK_SET_ERR(rval, "Could not get volumes from GTT");
+
+  return MB_SUCCESS;
+}
+
+// sets up the obb tree for the problem
+ErrorCode DagMC::setup_obbs() {
+  ErrorCode rval;
+
+  // If we havent got an OBB Tree, build one.
+  if (!GTT->have_obb_tree()) {
+    std::cout << "Building OBB Tree..." << std::endl;
+    rval = GTT->construct_obb_trees();
+    MB_CHK_SET_ERR(rval, "Failed to build obb trees");
+  }
+  return MB_SUCCESS;
+}
+
+// setups of the indices for the problem, builds a list of
+ErrorCode DagMC::setup_indices() {
+  Range surfs, vols;
+  ErrorCode rval = setup_geometry(surfs, vols);
+
+  // build the various index vectors used for efficiency
+  rval = build_indices(surfs, vols);
+  MB_CHK_SET_ERR(rval, "Failed to build surface/volume indices");
+  return MB_SUCCESS;
+}
+
+// initialise the obb tree
+ErrorCode DagMC::init_OBBTree() {
+  ErrorCode rval;
+
+  // find all geometry sets
+  rval = GTT->find_geomsets();
+  MB_CHK_SET_ERR(rval, "GeomTopoTool could not find the geometry sets");
+
+  // implicit compliment
+  // EntityHandle implicit_complement;
```

Comment:
Recommend deleting commented out code.  Optionally, you can leave an additional comment labeling under what condition the code can be re-instated or deleted.